flake.nix

```diff
@@ -13,77 +12,91 @@
+        */
+        default = self.packages.${system}.helix;
+      })
+      pkgsFor;
+    checks =
+      lib.mapAttrs (system: pkgs: let
+        # Get Helix's MSRV toolchain to build with by default.
+        msrvToolchain = pkgsFor.${system}.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+        msrvPlatform = pkgsFor.${system}.makeRustPlatform {
+          cargo = msrvToolchain;
+          rustc = msrvToolchain;
+        };
+      in {
+        helix = self.packages.${system}.helix.override {
+          buildType = "debug";
+          rustPlatform = msrvPlatform;
+        };
+      })
+      pkgsFor;
 
-      checks.helix = self.outputs.packages.${system}.helix.override {
-        buildType = "debug";
-        rustPlatform = msrvPlatform;
-      };
+    # Devshell behavior is preserved.
+    devShells =
+      lib.mapAttrs (system: pkgs: {
+        default = let
+          commonRustFlagsEnv = "-C link-arg=-fuse-ld=lld -C target-cpu=native --cfg tokio_unstable";
+          platformRustFlagsEnv = lib.optionalString pkgs.stdenv.isLinux "-Clink-arg=-Wl,--no-rosegment";
+        in
+          pkgs.mkShell {
+            inputsFrom = [self.checks.${system}.helix];
+            nativeBuildInputs = with pkgs;
+              [
+                lld
+                cargo-flamegraph
+                rust-bin.nightly.latest.rust-analyzer
+              ]
+              ++ (lib.optional (stdenv.isx86_64 && stdenv.isLinux) cargo-tarpaulin)
+              ++ (lib.optional stdenv.isLinux lldb)
+              ++ (lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.CoreFoundation);
+            shellHook = ''
+              export RUST_BACKTRACE="1"
+              export RUSTFLAGS="''${RUSTFLAGS:-""} ${commonRustFlagsEnv} ${platformRustFlagsEnv}"
+            '';
+          };
+      })
+      pkgsFor;
 
-      # Devshell behavior is preserved.
-      devShells.default = let
-        commonRustFlagsEnv = "-C link-arg=-fuse-ld=lld -C target-cpu=native --cfg tokio_unstable";
-        platformRustFlagsEnv = pkgs.lib.optionalString pkgs.stdenv.isLinux "-Clink-arg=-Wl,--no-rosegment";
-      in
-        pkgs.mkShell
-        {
-          inputsFrom = [self.checks.${system}.helix];
-          nativeBuildInputs = with pkgs;
-            [
-              lld
-              cargo-flamegraph
-              rust-bin.nightly.latest.rust-analyzer
-            ]
-            ++ (lib.optional (stdenv.isx86_64 && stdenv.isLinux) cargo-tarpaulin)
-            ++ (lib.optional stdenv.isLinux lldb)
-            ++ (lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.CoreFoundation);
-          shellHook = ''
-            export RUST_BACKTRACE="1"
-            export RUSTFLAGS="''${RUSTFLAGS:-""} ${commonRustFlagsEnv} ${platformRustFlagsEnv}"
-          '';
-        };
-    })
-    // {
-      overlays.default = final: prev: {
-        helix = final.callPackage ./default.nix {inherit gitRev;};
-      };
+    overlays.default = final: prev: {
```

Comment:
You may consider also providing `overlays.helix` and `default = self.overlays.helix`, to mirror `packages`. Not critical.